test/linter/main.go

```diff
@@ -49,6 +54,57 @@ func checkPath(filePath string, info os.FileInfo, err error) error {
 	return nil
 }
 
+func checkManifests(filePath string, info os.FileInfo, err error) error {
+	if err != nil {
+		return err
+	}
+	if path.Ext(filePath) == ".yaml" {
+		// Check if the path is ignored
+		for _, ignore := range ignoreManifestsSubPaths {
+			re, err := regexp.Compile(fmt.Sprintf("^%s$", ignore))
+			if err != nil {
+				return errors.Wrapf(err, "failed to compile regex %s", ignore)
+			}
+			if re.Match([]byte(filePath)) {
+				return nil
+			}
+		}
+		err := generateUniqueKeys(filePath)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Generates keys in form of kind-name-namespace or kind-name returns error in case key already exists
+func generateUniqueKeys(manifest string) error {
+	yamlFile, err := ioutil.ReadFile(manifest)
+	if err != nil {
+		return errors.Errorf("error reading the file %v", manifest)
+	}
+	manifestData, err := kommons.GetUnstructuredObjects(yamlFile)
+	if err != nil {
+		log.Warnf("error parsing the yaml %v", manifest)
+	}
+	for i := range manifestData {
+		if manifestData[i].Object["metadata"].(map[string]interface{})["namespace"] != nil {
+			value := fmt.Sprintf("%v-%v-%v", manifestData[i].Object["kind"].(string), manifestData[i].Object["metadata"].(map[string]interface{})["name"].(string), manifestData[i].Object["metadata"].(map[string]interface{})["namespace"].(string))
+			if !contains(allowedDuplicateKeys, value) && contains(keys, value) {
+				return errors.Errorf("error %v already present in the manifests", value)
```

Comment:
Could the error also state which file the duplicate is encountered in?

Reply:
Thinking of a way it could be possible.

Reply:
Changed keys to a map with map-key as the generated key and value as the manifest file. You can test by edit the allowedDuplicateManifests var to any other value and see the test fails with giving msg like:
```
FATA[0000] Manifest linting failed: error: key CustomResourceDefinition-servicemonitors.monitoring.coreos.com from manifests/crd/service-monitor.yaml already present in manifests/crd/prometheus.yaml
```

Reply:
Thanks, that is a nice level of verbosity